giraffe/src/utils/legend/sort.ts

```diff
@@ -0,0 +1,30 @@
+import {DomainLabel, LineData, LinePosition} from '../../types'
+
+export const getDataSortOrder = (
+  lineData: LineData,
+  rowIndices: number[],
+  position: LinePosition,
+  domainLabel?: DomainLabel
+): number[] => {
+  if (!position || position === 'overlaid') {
+    return rowIndices
+  }
+
+  const domainLabelName = domainLabel || DomainLabel.Y
+
+  const dataMap = {}
+  const measurementValues = Object.keys(lineData).reduce(
+    (accumulator, id) => accumulator.concat(lineData[id][domainLabelName]),
+    []
+  )
+  const sortable = []
+  rowIndices.forEach(hoveredRowIndex => {
+    if (!dataMap[measurementValues[hoveredRowIndex]]) {
```

Comment:
what does `measurementValues[hoveredRowIndex]` represent or mean? it might make this function easier to follow if that gets stored into a variable. It'll contextually what `dataMap` is actually mapping

Reply:
Yep, good call. I think originally I just couldn't come up with good names because I had thought they could be both strings or numbers possibly. But in thinking about how this is used, these are clearly numbers only.

Changed a bunch of names to `numericalValues`, `rowIndex`, `numberMap`, `combinedNumericalValues`. I think it makes the whole function a lot more readable and easier to reason about.